receiver/elasticsearchreceiver/metadata.yaml

```diff
@@ -483,9 +499,228 @@ metrics:
+      value_type: int
+    attributes: [ cluster_state_update_type ]
+    enabled: true
+  elasticsearch.node.cluster.state_update.commit_time:
+    description: The cumulative amount of time spent waiting for a cluster state update to commit.
+    unit: 1
+    sum:
+      monotonic: true
+      aggregation: cumulative
+      value_type: int
+    attributes: [ cluster_state_update_type ]
+    enabled: true
+  elasticsearch.node.cluster.state_update.completion_time:
+    description: The cumulative amount of time spent waiting for a cluster state update to complete.
+    unit: 1
+    sum:
+      monotonic: true
+      aggregation: cumulative
+      value_type: int
+    attributes: [ cluster_state_update_type ]
+    enabled: true
+  elasticsearch.node.cluster.state_update.master_apply_time:
+    description: The cumulative amount of time spent applying cluster state updates on the elected master since the node started.
+    unit: 1
+    sum:
+      monotonic: true
+      aggregation: cumulative
+      value_type: int
+    attributes: [ cluster_state_update_type ]
+    enabled: true
+  elasticsearch.node.cluster.state_update.notification_time:
+    description: The cumulative amount of time spent notifying listeners of a cluster state update since the node started.
+    unit: 1
+    sum:
+      monotonic: true
+      aggregation: cumulative
+      value_type: int
+    attributes: [ cluster_state_update_type ]
+    enabled: true
+  elasticsearch.node.ingest.count:
+    description: Total number of documents ingested during the lifetime of this node.
+    unit: "{documents}"
+    sum:
+      monotonic: true
+      aggregation: cumulative
+      value_type: int
+    attributes: [ ]
+    enabled: true
+  elasticsearch.node.ingest.current:
+    description: Total number of documents currently being ingested.
+    unit: "{documents}"
+    gauge:
+      value_type: int
+    attributes: [ ]
+    enabled: true
+  elasticsearch.node.ingest.failed:
+    description: Total number of failed ingest operations during the lifetime of this node.
+    unit: "{operation}"
+    sum:
+      monotonic: true
+      aggregation: cumulative
+      value_type: int
+    attributes: [ ]
+    enabled: true
+  elasticsearch.node.ingest.pipeline.count:
+    description: Total number of documents ingested during the lifetime of this node.
+    unit: "{documents}"
+    sum:
+      monotonic: true
+      aggregation: cumulative
+      value_type: int
+    attributes: [ ingest_pipeline_name ]
+    enabled: true
+  elasticsearch.node.ingest.pipeline.failed:
+    description: Total number of failed operations for the ingest pipeline.
+    unit: "{operation}"
+    sum:
+      monotonic: true
+      aggregation: cumulative
+      value_type: int
+    attributes: [ ingest_pipeline_name ]
+    enabled: true
+  elasticsearch.node.ingest.pipeline.current:
+    description: Total number of documents currently being ingested by a pipeline.
+    unit: "{documents}"
+    gauge:
```

Comment:
Non-monotonic sum?

Reply:
I believe gauge is the proper metric type here. Based on this definition:

> A *gauge* is a metric that represents a single numerical value that can arbitrarily go up and down.
> 
> Gauges are typically used for measured values like temperatures or current memory usage, but also "counts" that can go up and down, like the number of concurrent requests.

an aggregation of `elasticsearch.node.ingest.pipeline.current` provides no meaningful value.

Reply:
[Here is the open telemetry definition of a gauge](https://github.com/open-telemetry/opentelemetry-proto/blob/c5c8b28012583fda55b0cb16f73a820722171d49/opentelemetry/proto/metrics/v1/metrics.proto#L232). 

I think the question of which type of use in this case comes down to whether or not aggregating instances of this metric would provide a meaningful sum. The case for this would be for a user that is managing a fleet of pipelines, and wants to know the total number of documents being ingested across them.

A truly meaningless would be something like a sum of speeds or temperatures.

Reply:
Ah I see. Thanks for the explanation!

There are three metrics that share the common ` elasticsearch.node.pipeline.ingest` name:
- `elasticsearch.node.pipeline.ingest.documents.preprocessed`
- `elasticsearch.node.pipeline.ingest.operations.failed`
- `elasticsearch.node.pipeline.ingest.documents.current`

All three represents different metrics in my eyes. I guess we could combine `elasticsearch.node.pipeline.ingest.documents.preprocessed` and `elasticsearch.node.pipeline.ingest.documents.current` though I'm not sure if 'proprocessed' documents is another way of saying ES has ingested the document. What do you think?

Reply:
EDIT: I think I now understand what you meant by 'aggregating instances of this metric'.